### ISSUE TITLE
Potential error in apply_ufunc docstring for input_core_dims

### ISSUE THREAD

**Issue by nbren12** [pydata/xarray#1577]
The documentation for `input_core_dims` reads:
```
    input_core_dims : Sequence[Sequence], optional
        List of the same length as ``args`` giving the list of core dimensions
        on each input argument that should be broadcast. By default, we assume
        there are no core dimensions on any input arguments.

        For example ,``input_core_dims=[[], ['time']]`` indicates that all
        dimensions on the first argument and all dimensions other than 'time'
        on the second argument should be broadcast.
```

The first and second paragraphs seem contradictory to me. Shouldn't the first paragraph be changed to:

```
        List of the same length as ``args`` giving the list of core dimensions
        on each input argument that should *not* be broadcast. By default, we assume
        there are no core dimensions on any input arguments.
```

**Comment by nbren12** [comment]
Also, what it could be clarified what is meant by a `core` dimension.

**Comment by shoyer** [comment]
Yes, this was a typo. I noticed this the other day and fixed on one of my local branches -- it should definitely go in before we make this public API.

I'll see about clarifying the concept of "core dimensions". I think it makes sense once you read the numpy docs, but this should stand alone. It does really just mean "dimensions that shouldn't be used for broadcasting" but an example would probably help.

**Comment by nbren12** [comment]
My understanding is that the "core dimensions" are moved to the end of each input, and then singleton axes are inserted to make them broadcastable in a numpy sense. Is that correct?

**Comment by shoyer** [comment]
> My understanding is that the "core dimensions" are moved to the end of each input, and then singleton axes are inserted to make them broadcastable in a numpy sense. Is that correct?

Yes, exactly.

Though perhaps we should actually insert non-singleton dimensions to make it easier to understand.

**Comment by shoyer** [comment]
Fixed by #1517.
